### ISSUE TITLE
Switch testing over to tape/coverify

### ISSUE THREAD

**Issue by mikolalysenko** [scijs/ndarray-ldl-factorization#1]
The main advantage to this is that it is easier to use it in a browser and doesn't require any special test running harness.  Also it would be more consistent with the rest of the scijs ecosystem, which is currently using tape:

https://github.com/substack/tape

https://www.npmjs.com/package/coverify

**Comment by Planeshifter** [comment]
Thanks for the primers, will make myself familiar with those two packages.

**Comment by rreusser** [comment]
Yeah, I was debating what to use for testing. I got oriented with tape but couldn't ultimately find a reason to favor one over the other. Found testling but looked like that was bafflingly defunct.otherwise figured any testing at all was the main concern.

**Comment by mikolalysenko** [comment]
I've sort of converged on tape after a few years of working in node.  In terms of interface, it works identically to [tap](https://github.com/isaacs/node-tap), but has the benefit of working seamlessly in a browser.  I find other test harnesses have problems when you are testing across different environments, or output very complicated formats.  [The test anything protocol](https://testanything.org/) is probably the oldest test standard has the broadest support for the most number of tools.

**Comment by mikolalysenko** [comment]
Also another good tip is travisify, which makes CI a bit easier:

https://github.com/substack/travisify

**Comment by Planeshifter** [comment]
Thanks for the links! Although it is a bit off-topic, I was also wondering whether some helper functions were available for ndarrays although I couldn't find them in the list of ndarray modules: For example, a function to create an identity or diagonal matrix, similar to numpy's `diag` or `eye` functions?

**Comment by Planeshifter** [comment]
Made the switch to tape + coverify. `npm test` now executes those. I am not sure whether I prefer them to chai, mocha and coveralls tbh, but having a consistent collection should  supersede personal preference.

**Comment by rreusser** [comment]
I'm flexible but find the tap semi-human-readable-output moderately unpleasant. tap-spec helped. At the same time, the rspec `expect.this.to.be.something.with.too.many.chained.words` format is too far in the other direction. So yeah, on board with consistency. An ndarray-test-helpers module would help so we don't have to rewrite matrix and vector comparison functions fifty times.

**Comment by rreusser** [comment]
Ah, just found ndarray-show… this is great… should modify that to highlight mismatched matrix entries in red… while testing… but maybe that's unnecessary… Either way, I'm encouraged. It's proving pretty easy to keep hammering out modules. BLAS Level 2 (complex) and BLAS Level 3 are next so that I can attack householder QR.

**Comment by rreusser** [comment]
Still needs coverify.
